osu.Game/OsuGame.cs

```diff
@@ -504,7 +504,16 @@ private void loadComponentSingleFile<T>(T d, Action<T> add)
             // schedule is here to ensure that all component loads are done after LoadComplete is run (and thus all dependencies are cached).
             // with some better organisation of LoadComplete to do construction and dependency caching in one step, followed by calls to loadComponentSingleFile,
             // we could avoid the need for scheduling altogether.
-            Schedule(() => { asyncLoadStream = asyncLoadStream?.ContinueWith(t => LoadComponentAsync(d, add).Wait()) ?? LoadComponentAsync(d, add); });
+            Schedule(() => { asyncLoadStream = asyncLoadStream?.ContinueWith(async t =>
+            {
+                try
+                {
+                    await LoadComponentAsync(d, add);
+                }
+                catch (OperationCanceledException)
+                {
+                }
+            }) ?? LoadComponentAsync(d, add); });
```

Comment:
Can this be split out into multiple lines please? Super hard to decipher what `?? LoadComponentAsync(d, add)` was doing..

```
Schedule(() =>
{
    if (asyncLoadStream == null)
        asyncLoadStream = LoadComponentAsync(d, add);
    else
    {
        asyncLoadStream = asyncLoadStream?.ContinueWith(async t =>
        {
            try
            {
                await LoadComponentAsync(d, add);
            }
            catch (OperationCanceledException)
            {
            }
        });
    }
});
```